cmd/ltkeycloak/from_mattermost.go

```diff
@@ -0,0 +1,286 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sync"
+	"time"
+
+	gocloak "github.com/Nerzal/gocloak/v13"
+	"github.com/mattermost/mattermost-load-test-ng/deployment"
+	"github.com/mattermost/mattermost-load-test-ng/deployment/terraform"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/spf13/cobra"
+)
+
+func userTxtWriter(usersTxtChan chan string, doneChan chan struct{}, userPassword string) {
+	usersTxtFile, err := os.OpenFile("users.txt", os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		mlog.Error("failed to open users.txt file")
+	}
+	defer usersTxtFile.Close()
+
+	for {
+		select {
+		case email := <-usersTxtChan:
+			_, err := usersTxtFile.Write([]byte(model.UserAuthServiceSaml + ":" + email + " " + userPassword + "\n"))
+			if err != nil {
+				mlog.Error("failed to write to users.txt", mlog.String("err", err.Error()))
+			}
+
+		case <-doneChan:
+			return
+		}
+	}
+}
+
+type workerConfig struct {
+	workerNumber     int
+	usersChan        chan *model.User
+	wg               *sync.WaitGroup
+	mmClient         *model.Client4
+	keycloakClient   *gocloak.GoCloak
+	keycloakToken    *gocloak.JWT
+	keycloakRealm    string
+	userPassword     string
+	deploymentConfig *deployment.Config
+	doneChan         chan struct{}
+}
+
+func migrateMattermostUsersToKeycloak(worker *workerConfig) {
+	refreshTokenTicker := time.NewTicker(30 * time.Second)
```

Comment:
Wow, do we need to refresh the token every 30 seconds?

Reply:
Yeah, it has very limited expiration time.